experimental/packages/opentelemetry-exporter-metrics-otlp-http/src/transformMetrics.ts

```diff
@@ -14,246 +14,218 @@
-      aggregationTemporality: toAggregationTemporality(metric),
+        metricData.descriptor.type === InstrumentType.COUNTER ||
+        metricData.descriptor.type === InstrumentType.OBSERVABLE_COUNTER,
+      aggregationTemporality: toAggregationTemporality(aggregationTemporality),
     };
-    if (metric.descriptor.valueType === ValueType.INT) {
-      metricCollector.intSum = result;
-    } else {
-      metricCollector.doubleSum = result;
-    }
-  } else if (metric.aggregator.kind === AggregatorKind.LAST_VALUE) {
-    const result = {
-      dataPoints: [toDataPoint(metric, startTime)],
-    };
-    if (metric.descriptor.valueType === ValueType.INT) {
-      metricCollector.intGauge = result;
-    } else {
-      metricCollector.doubleGauge = result;
+
+    if (
+      metricData.descriptor.type === InstrumentType.COUNTER ||
+      metricData.descriptor.type === InstrumentType.OBSERVABLE_COUNTER ||
+      metricData.descriptor.type === InstrumentType.UP_DOWN_COUNTER ||
+      metricData.descriptor.type === InstrumentType.OBSERVABLE_UP_DOWN_COUNTER
+    ) {
+      if (metricData.descriptor.valueType === ValueType.INT) {
+        metricCollector.intSum = result;
+      } else {
+        metricCollector.doubleSum = result;
+      }
+    } else{
+      // Instrument is a gauge.
+      if (metricData.descriptor.valueType === ValueType.INT) {
+        metricCollector.intGauge = result;
+      } else {
+        metricCollector.doubleGauge = result;
+      }
     }
-  } else if (metric.aggregator.kind === AggregatorKind.HISTOGRAM) {
+  } else if (metricData.dataPointType === DataPointType.HISTOGRAM) {
     const result = {
-      dataPoints: [toHistogramPoint(metric, startTime)],
-      aggregationTemporality: toAggregationTemporality(metric),
+      dataPoints: toHistogramDataPoints(metricData),
+      aggregationTemporality: toAggregationTemporality(aggregationTemporality)
     };
-    if (metric.descriptor.valueType === ValueType.INT) {
+    if (metricData.descriptor.valueType === ValueType.INT) {
       metricCollector.intHistogram = result;
     } else {
       metricCollector.doubleHistogram = result;
     }
   }
 
+  // TODO: Add support for exponential histograms when they're ready.
+
   return metricCollector;
 }
 
 /**
  * Prepares metric service request to be sent to collector
  * @param metrics metrics
- * @param startTime start time of the metric in nanoseconds
- * @param collectorMetricExporterBase
+ * @param aggregationTemporality
+ * @param collectorExporterBase
  */
-export function toOTLPExportMetricServiceRequest<
-  T extends otlpTypes.OTLPExporterConfigBase
->(
-  metrics: MetricRecord[],
-  startTime: number,
-  collectorExporterBase: OTLPExporterBase<
-    T,
-    MetricRecord,
-    otlpTypes.opentelemetryProto.collector.metrics.v1.ExportMetricsServiceRequest
-  >
+export function toOTLPExportMetricServiceRequest<T extends otlpTypes.OTLPExporterConfigBase>(
```

Comment:
I'll bring up my previous question: since the protos used here are somewhat outdated, should the transforms done in exporters use the updated proto transformer package? I'm not sure what is the final requirement for #2774, but this transformation code can be get rid of then.

Reply:
Thanks for bringing this up. :slightly_smiling_face: I changed the description so that #2774 will not be closed after this PR. I will do that in a follow-up PR to unblock everyone who is currently waiting on this PR. I added a comment to the issue as well. https://github.com/open-telemetry/opentelemetry-js/issues/2774#issuecomment-1088375123